examples/sign_verify.rs

```diff
@@ -0,0 +1,35 @@
+extern crate bitcoin_hashes;
+extern crate secp256k1;
+
+use bitcoin_hashes::{sha256, Hash};
+use secp256k1::{All, Error, Message, PublicKey, Secp256k1, SecretKey, Signature};
+
+fn verify(secp: &Secp256k1<All>, msg: &[u8], sig: [u8; 64], pubkey: [u8; 33]) -> Result<bool, Error> {
```

Comment:
I really think we should parameterize the `Secp2561` over `C: Verification` rather than forcing it to be `All`. This is more flexible, provides better documentation, and more accurately reflects good practice.

Reply:
Right. I didn't want to introduce generics for the example but I guess you're right.